ome/resource_discovery_schema.go

```diff
@@ -0,0 +1,469 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// DiscoveryJobSchema for discovery job schema.
+func DiscoveryJobSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+
+		"discovery_job_id": schema.Int64Attribute{
+			MarkdownDescription: "ID of the discovery configuration group",
+			Description:         "ID of the discovery configuration group",
+			Computed:            true,
+		},
+
+		"discovery_job_name": schema.StringAttribute{
+			MarkdownDescription: "Name of the discovery configuration job",
+			Description:         "Name of the discovery configuration job",
+			Required:            true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+			},
+		},
+
+		"email_recipient": schema.StringAttribute{
+			MarkdownDescription: `
+				- Enter the email address to which notifications are to be sent about the discovery job status.
+				- Configure the SMTP settings to allow sending notifications to an email address.`,
+			Description: `
+				- Enter the email address to which notifications are to be sent about the discovery job status.
+				- Configure the SMTP settings to allow sending notifications to an email address.`,
+			Optional: true,
+			Computed: true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+			},
+		},
+
+		"discovery_config_targets": schema.SetNestedAttribute{
+			MarkdownDescription: `
+				- Provide the list of discovery targets.
+      			- Each discovery target is a set of \"network_address_detail\", \"device_types\", and one or more protocol credentials.`,
+			Description: `
+				- Provide the list of discovery targets.
+      			- Each discovery target is a set of \"network_address_detail\", \"device_types\", and one or more protocol credentials.`,
+			Required:     true,
+			NestedObject: schema.NestedAttributeObject{Attributes: DiscoveryConfigTargetsSchema()},
+		},
+
+		"job_wait": schema.BoolAttribute{
+			MarkdownDescription: "Provides the option to wait for job completion",
+			Description:         "Provides the option to wait for job completion",
+			Optional:            true,
+			Computed:            true,
+			PlanModifiers: []planmodifier.Bool{
+				BoolDefaultValue(types.BoolValue(true)),
+			},
+		},
+
+		"job_wait_timeout": schema.Int64Attribute{
+			MarkdownDescription: "The maximum wait time of job_wait in seconds. The job is tracked only for this duration.",
+			Description:         "The maximum wait time of job_wait in seconds. The job is tracked only for this duration.",
+			Optional:            true,
+			Computed:            true,
+			PlanModifiers: []planmodifier.Int64{
+				Int64DefaultValue(types.Int64Value(10800)),
+			},
+		},
+
+		"schedule": schema.StringAttribute{
```

Comment:
Can't this just be a bool attribute?

Reply:
we are keeping schedule as string attribute with two choices runnow and runlater